pkg/server/server.go

```diff
@@ -150,16 +154,70 @@ func (s *Server) ListenAndServe(ctx context.Context, logger log.Logger, port str
 		return fmt.Errorf("failed to register pprof handlers: %w", err)
 	}
 
+	// Strip the subpath
 	uiFS, err := fs.Sub(ui.FS, "packages/app/web/dist")
 	if err != nil {
 		return fmt.Errorf("failed to initialize UI filesystem: %w", err)
 	}
 
+	// There is currently no way to go between `http.FileServer(http.FS(uiFS))` and execute
+	// templates an http.ServeMux with registered paths and executes templates is used.
+	uiHandler := http.ServeMux{}
+	err = fs.WalkDir(uiFS, ".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if d.IsDir() {
+			return nil
+		}
+
+		b, err := fs.ReadFile(uiFS, path)
+
+		if err != nil {
+			return fmt.Errorf("failed to read ui file %s: %w", path, err)
+		}
+
+		tmpl, err := template.New(path).Parse(string(b))
+
+		if err != nil {
+			return fmt.Errorf("failed to parse ui file %s: %w", path, err)
+		}
+
+		var outputBuffer bytes.Buffer
+
+		err = tmpl.Execute(&outputBuffer, s.uiData)
+
+		if err != nil {
+			return fmt.Errorf("failed to execute ui file %s: %w", path, err)
+		}
+
+		outputBytes := outputBuffer.Bytes()
+
+		paths := []string{fmt.Sprintf("/%s", path)}
+
+		if paths[0] == "/index.html" {
+			paths = append(paths, "/")
+		}
+
+		for _, path := range paths {
+			uiHandler.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
+				http.ServeContent(w, r, d.Name(), time.Now(), bytes.NewReader(outputBytes)) // TODO(modtime) ? buildtime
```

Comment:
here is one todo left -> we could pass down the time of the build here, for client caching.

Reply:
Nice idea, however, I'd actually be in favor of removing build time altogether. 
Build times aren't good for reproducible builds.
I've opened an issue here: https://github.com/parca-dev/parca/issues/468

Reply:
:+1: for pointing that out - dropped also the `date` from the `--version` flag